### PR TITLE
feat: add emergencyOrPoolAdmin to freeze on aave v2

### PR DESCRIPTION
**Pull request by brotherlymite**

### CODE REVIEW

File: src/v2AmmEthPoolConfigurator/LendingPoolConfigurator/contracts/protocol/libraries/helpers/Errors.sol [hunk 1]
```diff
@@ -103,6 +103,7 @@ library Errors {
   string public constant LP_NOT_CONTRACT = '78';
   string public constant SDT_STABLE_DEBT_OVERFLOW = '79';
   string public constant SDT_BURN_EXCEEDS_BALANCE = '80';
+  string public constant CALLER_NOT_POOL_OR_EMERGENCY_ADMIN = '81'; // 'The caller must be the emergency or pool admin'
```
Comment (eboadom) [inline src/v2AmmEthPoolConfigurator/LendingPoolConfigurator/contracts/protocol/libraries/helpers/Errors.sol:106]: Mind that `81` is already taken for the stable debt token upgrade, and `82` for the one of collateral manager.
So should be `83`.

Also, don't we use a prefix for the LendingPoolConfigurator?

Comment (brotherlymite) [reply src/v2AmmEthPoolConfigurator/LendingPoolConfigurator/contracts/protocol/libraries/helpers/Errors.sol]: Fixed

File: tests/V2PoolConfiguratorTestBase.t.sol [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  function test_reverts_unfreezeReserve(address caller) public {
+    vm.assume(caller != EMERGENCY_ADMIN && caller != POOL_ADMIN);
+
+    vm.expectRevert(bytes(Errors.LPC_CALLER_NOT_POOL_OR_EMERGENCY_ADMIN));
+    vm.startPrank(caller);
+    ILendingPoolConfigurator(POOL_CONFIGURATOR).unfreezeReserve(UNDERLYING_ASSET);
+    vm.stopPrank();
+  }
+
+  function test_freezeReserve_emergencyAdmin() public {
```
Comment (miguelmtzinf) [inline tests/V2PoolConfiguratorTestBase.t.sol:53]: Just a note, `freeze`/`unfreeze` sets the state without checking the previous one. The following tests are checking the state set after the function call. It's possible that the final state check passes due to a previous state (while execution not working well).
No biggie, functions seem to work well, but if you wanna be more accurate you can record writes of the contract.

Comment (brotherlymite) [reply tests/V2PoolConfiguratorTestBase.t.sol:56]: Makes sense, fixed the tests to check the before and after state.

**Review by miguelmtzinf — APPROVED** [review]

**Review by eboadom — APPROVED** [review]

**Pull request merged**